generation/earl-grey.js

```diff
@@ -0,0 +1,167 @@
+const t = require("babel-types");
+const objectiveCParser = require("objective-c-parser");
+const generate = require("babel-generator").default;
+const fs = require("fs");
+const {
+  isNumber,
+  isString,
+  isBoolean,
+  isPoint,
+  isOneOf,
+} = require('./type-checks');
+
+function createClass(json) {
```

Comment:
Would love to see some comment code on what these json structures look like, or at least references to them somewhere. It helps when visualising things moving through the functions 👍